experiments/unbinned_spectrum_dataset/unbinned_datasets.py

```diff
@@ -0,0 +1,291 @@
+        """Make list of events in the ON  region.
+
+        Parameters
+        ----------
+        region : `~regions.SkyRegion`
+            Region to compute counts spectrum for.
+        observation: `~gammapy.data.DataStoreObservation`
+            Observation to compute effective area for.
+
+        Returns
+        -------
+        events : `~gammapy.data.event_list.EventList`
+            list of ON events
+        """
+        events = observation.events.select_region(region, wcs=self.geom_ref(region).wcs)
+        return events
+
+    def run(self, dataset, observation):
+        """Make unbinned spectrum dataset.
+
+        Parameters
+        ----------
+        dataset : `~gammapy.spectrum.SpectrumDataset`
+            Spectrum dataset.
+        observation: `~gammapy.data.DataStoreObservation`
+            Observation to reduce.
+
+        Returns
+        -------
+        dataset : `~gammapy.spectrum.SpectrumDataset`
+            Spectrum dataset.
+        """
+        kwargs = {
+            "name": f"{observation.obs_id}",
+            "gti": observation.gti,
+            "livetime": observation.observation_live_time_duration,
+        }
+        energy_axis = dataset.counts.energy
+        energy_axis_true = dataset.aeff.data.axis("energy")
+        region = dataset.counts.region
+
+        if "events" in self.selection:
+            kwargs["events"] = self.make_events(region, observation)
+            # array of dummy counts, not filled, the on region will be searched
+            # in here by the ReflectedRegionsBackgroundMaker, so we keep it
+            kwargs["counts"] = self.make_counts(region, energy_axis, observation)
+
+        if "aeff" in self.selection:
+            kwargs["aeff"] = self.make_aeff(region, energy_axis_true, observation)
+
+        if "edisp" in self.selection:
+            kwargs["edisp"] = self.make_edisp(
+                region.center, energy_axis, energy_axis_true, observation
+            )
+
+        return UnbinnedSpectrumDataset(**kwargs)
+
+
+class UnbinnedSpectrumDataset(SpectrumDataset):
+    """Unbinned pectrum dataset for on-off likelihood fitting."""
+
+    def __init__(
+        self,
+        models=None,
+        events=None,
+        counts=None,
+        livetime=None,
+        aeff=None,
+        edisp=None,
+        background=None,
+        mask_safe=None,
+        mask_fit=None,
+        name="",
+        gti=None,
+    ):
+        SpectrumDataset.__init__(
+            self,
+            models=models,
+            counts=counts,
+            livetime=livetime,
+            aeff=aeff,
+            edisp=edisp,
+            background=background,
+            mask_safe=mask_safe,
+            mask_fit=mask_fit,
+            name="",
+            gti=gti,
+        )
+        # events in the signal region
+        self.events = events
+
+
+class UnbinnedSpectrumDatasetOnOff(UnbinnedSpectrumDataset):
+    """The UnbinnedSpectrumDatasetOnOff has the events_off property
```

Comment:
Why not inherit from `SpectrumDatasetOnOff` directly? 
Then you don't need the `UnbinnedSpectrumDataset`.
You create a regular `SpectrumDataset` with the `SpectrumDatasetMaker` and then your modified `ReflectedRegionsBackgroundMaker` can create the `UnbinnedSpectrumDatasetOnOff` (it is long to write :)